module/VuFind/src/VuFind/Db/Service/UserListService.php

```diff
@@ -0,0 +1,414 @@
+     */
+    public function __construct(
+        EntityManager $entityManager,
+        EntityPluginManager $entityPluginManager,
+        Tags $tagParser,
+        Container $session = null
+    ) {
+        parent::__construct($entityManager, $entityPluginManager);
+        $this->tagParser = $tagParser;
+        $this->session = $session;
+    }
+
+    /**
+     * Get an array of resource tags associated with the list.
+     *
+     * @param UserList $list UserList object.
+     *
+     * @return array
+     */
+    public function getResourceTags($list)
+    {
+        $user = $list->getUser();
+        $tags = $this->getDbService(\VuFind\Db\Service\TagService::class)
+            ->getListTagsForUser($user, null, $list);
+        return $tags;
+    }
+
+    /**
+     * Create a userlist entity object.
+     *
+     * @return UserList
+     */
+    public function createUserList(): UserList
+    {
+        $class = $this->getEntityClass(UserList::class);
+        return new $class();
+    }
+
+    /**
+     * Create a new list object.
+     *
+     * @param User|bool $user User object representing owner of
+     * new list (or false if not logged in)
+     *
+     * @return UserList|bool
+     * @throws LoginRequiredException
+     */
+    public function getNew($user)
+    {
+        if (!$user) {
+            throw new LoginRequiredException('Log in to create lists.');
+        }
+        $user = is_object($user) ? $user : $this->entityManager->getReference(User::class, $user);
+        $row = $this->createUserList()
+            ->setCreated(new \DateTime())
+            ->setUser($user);
+        return $row;
+    }
+
+    /**
+     * Retrieve a list object.
+     *
+     * @param int $id Numeric ID for existing list.
+     *
+     * @return UserList
+     * @throws RecordMissingException
+     */
+    public function getExisting($id)
+    {
+        $result = $this->getEntityById(\VuFind\Db\Entity\UserList::class, $id);
+        if (empty($result)) {
+            throw new RecordMissingException('Cannot load list ' . $id);
+        }
+        return $result;
+    }
+
+    /**
+     * Update and save the list object using a request object -- useful for
+     * sharing form processing between multiple actions.
+     *
+     * @param User|int|bool              $user    Logged-in user (false if none)
+     * @param UserList                   $list    User list that is being modified
+     * @param \Laminas\Stdlib\Parameters $request Request to process
+     *
+     * @return int ID of newly created list
+     * @throws ListPermissionException
+     * @throws MissingFieldException
+     */
+    public function updateFromRequest($user, $list, $request)
+    {
+        $list->setTitle($request->get('title'))
+            ->setDescription($request->get('desc'))
+            ->setPublic((bool)$request->get('public'));
+
+        $this->save($list, $user);
+
+        $this->rememberLastUsed($list);
```

Comment:
`$this->save()` is calling `$this->rememberLastUsed`, so I don't think we need to call it again here.